### PR TITLE
New Tool - DR Curves using drc

### PR DESCRIPTION
**Pull request by rmassei**
Added a new tool for modelling DR curves fitting the best logistic or weibull model
**Output:**
- CSV File with different effect levels (drc)
- jpeg file with plotted DR curve (ggplot2)

### CODE REVIEW

File: tools/tox_tools/dose_responses/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+categories:
+  - Ecology
+  - Text Manipulation
+description: Toxicity prediction using QSAR models
+long_description: Toxicity prediction using QSAR models. Possible to select pre-defined models as well as user inputs.
+name: Dose Response Curve for Toxicological Risk Assessement
+owner: ufz
```
Comment (bernt-matthias) [inline tools/tox_tools/dose_responses/.shed.yml:7]: That's a good point. We would need an toolshed user `ufz`.

File: tools/tox_tools/dose_responses/dose_response.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        Rscript $__tool_directory__/dose_response.R
+            '$input_csv'
+            '$concentration_column'
+            '$response_column'
+            '$plot_output'
+            '$ec_output'
+        ]]>
+    </command>
+    <inputs>
+        <param name="input_csv" type="data" format="csv" label="Input CSV file"/>
```
Comment (bernt-matthias) [inline tools/tox_tools/dose_responses/dose_response.xml:23]: In general tab separated tables (`tabular`) are preferred in Galaxy. There are also converters (and I think csv is auto-converted).

File: tools/tox_tools/dose_responses/dose_response.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        <param name="concentration_column" type="text" label="Concentration Column" help="Name of the column for concentration values"/>
```
Comment (bernt-matthias) [inline tools/tox_tools/dose_responses/dose_response.xml:24]: You could use the `type="data_column"`  to specify a parameter that allow to select a column from `input_csv`. See https://docs.galaxyproject.org/en/master/dev/schema.html#data-column

File: tools/tox_tools/dose_responses/dose_response.xml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        <param name="response_column" type="text" label="Response Column" help="Name of the column for response values"/>
+    </inputs>
+    <outputs>
+        <data name="plot_output" format="jpeg" label="Dose Response Plot"/>
```
Comment (bernt-matthias) [inline tools/tox_tools/dose_responses/dose_response.xml:28]: Should be `jpg` (https://github.com/galaxyproject/galaxy/blob/3359a75ffc018f2d64dd93c654a4242543ab44d5/lib/galaxy/config/sample/datatypes_conf.xml.sample#L241C26-L241C29)

File: tools/tox_tools/dose_responses/dose_response.xml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        <data name="ec_output" format="csv" label="EC Values"/>
```
Comment (bernt-matthias) [inline tools/tox_tools/dose_responses/dose_response.xml:29]: I would prefer `label="${tool.name} on ${on_string}: EC Values"`

File: tools/tox_tools/dose_responses/dose_response.xml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    </outputs>
+            <tests>
```
Comment (bernt-matthias) [inline tools/tox_tools/dose_responses/dose_response.xml:31]: Indentation is off here.

**Review by bernt-matthias — COMMENTED** [review]
Thanks @rmassei  here a few comments. 

Could you also fix the errors reported by CI?

File: tools/tox_tools/dose_responses/.shed.yml (hunk 1 above)
Comment (bernt-matthias) [reply tools/tox_tools/dose_responses/.shed.yml:7]: For background the user belonging to this api key https://github.com/Helmholtz-UFZ/galaxy-tools/blob/e409f80039e6d17577b5c52b97a56d915f876822/.github/workflows/pr.yaml#L413 needs write access to the tool repo in the tool shed. 

I think having an UFZ user (`ufz`) would be a good idea.

File: tools/tox_tools/dose_responses/dose_response.xml [hunk 7]
```diff
@@ -0,0 +1,53 @@
+<tool id="dr_curve" name="Dose Response Curve for Toxicological Risk Assessement" version="3.0_1.galaxy0">
+    <description>Toxicity prediction tool -  Dose response Curve</description>
+    <creator>
+        <organization name="Helmholtz Centre for Environmental Research - UFZ, Department of Ecotoxicology"
+                      url ="https://www.ufz.de/index.php?en=34241"/>
+    </creator>
+    <requirements>
+        <requirement type="package" version="3.0_1">r-drc</requirement>
```
Comment (bernt-matthias) [inline tools/tox_tools/dose_responses/dose_response.xml:8]: Excellent. Or even better with macro tokens:

```suggestion
<tool id="dr_curve" name="Dose Response Curve for Toxicological Risk Assessment" version="@TOOL_VERSION@.galaxy@VERSION_SUFFIX@">
    <description>Toxicity prediction tool -  Dose response Curve</description>
    <macros>
        <token name="@TOOL_VERSION@">3.0_1</token>
        <token name="@VERSION_SUFFIX@">0</token>
    </macros>
    <creator>
        <organization name="Helmholtz Centre for Environmental Research - UFZ, Department of Ecotoxicology"
                      url ="https://www.ufz.de/index.php?en=34241"/>
    </creator>
    <requirements>
        <requirement type="package" version="@TOOL_VERSION@">r-drc</requirement>
```

File: tools/tox_tools/dose_responses/dose_response.xml [hunk 8]
```diff
@@ -0,0 +1,57 @@
+<tool id="dr_curve" name="Dose Response Curve for Toxicological Risk Assessment" version="@TOOL_VERSION@.galaxy@VERSION_SUFFIX@">
```
Comment (bernt-matthias) [inline tools/tox_tools/dose_responses/dose_response.xml:1]:
```suggestion
<tool id="dr_curve" name="Dose Response Curve for Toxicological Risk Assessment" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="23.0">
```

**Review by bernt-matthias — APPROVED** [review]
Hi @rmassei. This looks good to me. 

I will try to setup a user `ufz` at the tool shed and adapt out CI here. Small problem is with the other tools in here. The owner (i.e. my old TS user account) will need to allow write access to this tools for the `ufz` user. ... Despite this extra effort I think its a good idea.

FYI: If you want to install this tool (or any other) on the UFZ Galaxy you can just add it to this list: https://github.com/Helmholtz-UFZ/ufz-galaxy-tools/blob/main/tool_list.yaml

**Comment by rmassei** [comment]
Perfect, thanks for all the useful comments and improvements!

**Pull request merged**